CMakeFuncs.cmake

```diff
@@ -0,0 +1,627 @@
+# This is a set of common functions and settings for rippled
```

Comment:
PS. The complexity in this `CMakeFuncs.cmake` does make the `CMakeLists.txt` a lot simpler than it was before.